### PR TITLE
Add solana name service support

### PR DESCRIPTION
**Pull request by 0xavro**
Adding support for solana name service.
Currently Twitter handles take precedence over SNS and SNS takes precedence over the raw wallet public key.

Next step would be to add SWR support for fetching the SNS names.

Screen Shot 2022-05-18 at 11 48 24 PM

### CODE REVIEW

File: packages/dialect-react-ui/components/DisplayAddress/index.tsx [hunk 1]
```diff
@@ -81,6 +82,42 @@ const TwitterHandle = ({
...
+  return `${domain.slice(0, 4)}...${domain.slice(domain.length - 2)}`;
+};
+
+const SolanaName = ({
+  address,
+  displayName,
+  loadingName,
+  dimensionClassName = '',
+}: {
+  address: PublicKey | undefined;
```
Comment (Kiryous) [inline packages/dialect-react-ui/components/DisplayAddress/index.tsx:98]: coulde be `address?`

**Review by Kiryous — APPROVED** [review]
Looks great! Just one q in comments and also: should this PR add resolving domains on the CreateThread view?

File: packages/dialect-react-ui/components/DisplayAddress/index.tsx (hunk 1 above)
Comment (0xavro) [reply packages/dialect-react-ui/components/DisplayAddress/index.tsx:98]: yep let's do this instead

**Review by cbosborn — APPROVED** [review]
lgtm. let's get thread creation support in a follow up pr

**Pull request merged**